agent/command/s3_get.go

```diff
@@ -33,67 +30,21 @@ var (
 	s3GetInternalBucketAttribute       = fmt.Sprintf("%s.internal_bucket", s3GetAttribute)
 )
 
 // s3get is a command to fetch a resource from an S3 bucket and download it to
 // the local machine.
 type s3get struct {
-	// AwsKey, AwsSecret, and AwsSessionToken are the user's credentials for
-	// authenticating interactions with s3.
-	AwsKey          string `mapstructure:"aws_key" plugin:"expand"`
-	AwsSecret       string `mapstructure:"aws_secret" plugin:"expand"`
-	AwsSessionToken string `mapstructure:"aws_session_token" plugin:"expand"`
-
-	// RemoteFile is the file path of the file to get, within its bucket.
-	RemoteFile string `mapstructure:"remote_file" plugin:"expand"`
-
-	// remoteFile is the file path without any expansions applied.
-	remoteFile string
-
-	// Region is the S3 region where the bucket is located. It defaults to
-	// "us-east-1".
-	Region string `mapstructure:"region" plugin:"region"`
-
-	// Bucket is the S3 bucket holding the desired file.
-	Bucket string `mapstructure:"bucket" plugin:"expand"`
-
-	// BuildVariants stores a list of build variants to run the command for.
-	// If the list is empty, it runs for all build variants.
-	BuildVariants []string `mapstructure:"build_variants" plugin:"expand"`
-
 	// Only one of these two should be specified. local_file indicates that the
 	// s3 resource should be downloaded as-is to the specified file, and
 	// extract_to indicates that the remote resource is a .tgz file to be
 	// downloaded to the specified directory.
 	LocalFile string `mapstructure:"local_file" plugin:"expand"`
 	ExtractTo string `mapstructure:"extract_to" plugin:"expand"`
 
-	// Optional, when set to true, causes this command to be skipped over without an error when
-	// the path specified in remote_file does not exist. Defaults to false, which triggers errors
-	// for missing files.
-	Optional string `mapstructure:"optional" plugin:"expand"`
-
-	// TemporaryRoleARN is not meant to be used in production. It is used for testing purposes
-	// relating to the DEVPROD-5553 project.
-	// This is an ARN that should be assumed to make the S3 request.
-	// TODO (DEVPROD-13982): Upgrade this flag to RoleARN.
-	TemporaryRoleARN string `mapstructure:"temporary_role_arn" plugin:"expand"`
-
-	// TemporaryUseInternalBucket is not meant to be used in production. It is used for testing purposes
-	// relating to the DEVPROD-5553 project.
-	// This flag is used to determine if the s3_credentials route should be called before the command is executed.
-	// TODO (DEVPROD-13982): Remove this flag and use the internal bucket list to determine if the s3_credentials
-	// route should be called.
-	TemporaryUseInternalBucket string `mapstructure:"temporary_use_internal_bucket" plugin:"expand"`
-
-	skipMissing bool
-
-	bucket          pail.Bucket
-	internalBuckets []string
-
-	temporaryUseInternalBucket bool
-
+	s3Operation `plugin:"expand"`
```

Comment:
The the deleted above is moved inside s3Operation. This also has the struct tag so it can have expansions inside of it